src/main/java/VASSAL/build/module/Map.java

```diff
@@ -1232,15 +1232,17 @@ public void mouseExited(MouseEvent e) {
    */
   @Override
   public void mouseClicked(MouseEvent e) {
+    //BR// Bug13137 - don't write over java's mouse event
+    MouseEvent mapEvent = new MouseEvent(e.getComponent(), e.getID(), e.getWhen(), e.getModifiersEx(), e.getX(), e.getY(), e.getXOnScreen(), e.getYOnScreen(), e.getClickCount(), e.isPopupTrigger(), e.getButton());
     if (!mouseListenerStack.isEmpty()) {
-      final Point p = componentToMap(e.getPoint());
-      e.translatePoint(p.x - e.getX(), p.y - e.getY());
-      mouseListenerStack.get(mouseListenerStack.size()-1).mouseClicked(e);
+      final Point p = componentToMap(mapEvent.getPoint());
+      mapEvent.translatePoint(p.x - mapEvent.getX(), p.y - mapEvent.getY());
+      mouseListenerStack.get(mouseListenerStack.size()-1).mouseClicked(mapEvent);
     }
     else if (multicaster != null) {
-      final Point p = componentToMap(e.getPoint());
-      e.translatePoint(p.x - e.getX(), p.y - e.getY());
-      multicaster.mouseClicked(e);
+      final Point p = componentToMap(mapEvent.getPoint());
+      mapEvent.translatePoint(p.x - mapEvent.getX(), p.y - mapEvent.getY());
+      multicaster.mouseClicked(mapEvent);
     }
```

Comment:
You could eliminate quite a bit of repeated code here and below by adding a method which copies the event, does the translation, and returns the copied event.

Then you could do this:
```
public void mouseClicked(MouseEvent e) {
    if (!mouseListenerStack.isEmpty()) {
      mouseListenerStack.get(mouseListenerStack.size()-1).mouseClicked(translateEvent(e));
    }
    else if (multicaster != null) {
      multicaster.mouseClicked(translateEvent(e));
    }
  }
```
and a similar thing in `mousePressed()` and `mouseReleased()`.

Reply:
Fixed that up. Yan had mentioned that, but in a context where he was going to go do that in a second pass after the PR. But anyway now you've mentioned it and I've done it.